From a14bd210db66ecb4aecbb4975dd1d74f683f7453 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Harald=20Jens=C3=A5s?= <hjensas@redhat.com>
Date: Tue, 8 Mar 2022 22:32:27 +0100
Subject: [PATCH] Set agent_type in tests

Networking-Baremetal is already hard-coding the agent
type. The issue was that tests don't set agent type in
the Fake Port Context.

Also removes an override of force_config_drive to false
which is causing failures in jobs designed around use
of configuration drives.

Change-Id: I171845e556a45746d9435a11390e510c731f6d5c
Story: 2009898
Task: 44696
(cherry picked from commit 3f25f63578344dbba61c572614f9ccdedc12fb6e)
(cherry picked from commit 6bcfcaf5a136c28792e55259fabd7663bfdd00bc)
---
 .../tests/unit/plugins/ml2/test_baremetal_mech.py    | 12 ++++++++----
 zuul.d/networking-baremetal-jobs.yaml                |  1 -
 2 files changed, 8 insertions(+), 5 deletions(-)

diff --git a/networking_baremetal/tests/unit/plugins/ml2/test_baremetal_mech.py b/networking_baremetal/tests/unit/plugins/ml2/test_baremetal_mech.py
index 3088c2c..893f970 100644
--- a/networking_baremetal/tests/unit/plugins/ml2/test_baremetal_mech.py
+++ b/networking_baremetal/tests/unit/plugins/ml2/test_baremetal_mech.py
@@ -37,13 +37,17 @@ class TestBaremetalMechDriver(base.AgentMechanismBaseTestCase):
     BAD_CONFIGS = {
         'bridge_mappings': {'wrong_physical_network': 'wrong_physnet'}
     }
-    AGENTS = [{'alive': True, 'configurations': GOOD_CONFIGS, 'host': 'host'}]
+    AGENTS = [{'agent_type': AGENT_TYPE, 'alive': True,
+               'configurations': GOOD_CONFIGS, 'host': 'host'}]
     AGENTS_DEAD = [
-        {'alive': False, 'configurations': GOOD_CONFIGS, 'host': 'dead_host'}
+        {'agent_type': AGENT_TYPE, 'alive': False,
+         'configurations': GOOD_CONFIGS, 'host': 'dead_host'}
     ]
     AGENTS_BAD = [
-        {'alive': False, 'configurations': GOOD_CONFIGS, 'host': 'bad_host_1'},
-        {'alive': True, 'configurations': BAD_CONFIGS, 'host': 'bad_host_2'}
+        {'agent_type': AGENT_TYPE, 'alive': False,
+         'configurations': GOOD_CONFIGS, 'host': 'bad_host_1'},
+        {'agent_type': AGENT_TYPE, 'alive': True,
+         'configurations': BAD_CONFIGS, 'host': 'bad_host_2'}
     ]
     VNIC_TYPE = portbindings.VNIC_BAREMETAL
 
diff --git a/zuul.d/networking-baremetal-jobs.yaml b/zuul.d/networking-baremetal-jobs.yaml
index 98a5cd2..37d49f9 100644
--- a/zuul.d/networking-baremetal-jobs.yaml
+++ b/zuul.d/networking-baremetal-jobs.yaml
@@ -19,7 +19,6 @@
       devstack_localrc:
         BUILD_TIMEOUT: 2400
         ENABLE_TENANT_VLANS: True
-        FORCE_CONFIG_DRIVE: False
         IRONIC_DEFAULT_DEPLOY_INTERFACE: direct
         IRONIC_DEFAULT_RESCUE_INTERFACE: ""
         IRONIC_ENABLED_NETWORK_INTERFACES: flat,neutron
-- 
2.35.1

